doc/conf.py

```diff
@@ -140,6 +145,48 @@
 }
 
 
+# -- Options for HTML output ----------------------------------------------
+from sphinx.ext.napoleon import GoogleDocstring
+from sphinx.locale import _, __
+# rebind examples section to automate placement with our custom
+# .. pyvista-plot:: directive by mangling docstrings
+
+def _custom_parse_generic_section(self, section, use_admonition):
+    lines = self._strip_empty(self._consume_to_next_section())
+    lines = self._dedent(lines)
+    if use_admonition:
+        header = '.. admonition:: %s' % section
+        lines = self._indent(lines, 3)
+    else:
+        header = '.. rubric:: %s' % section
+
+    # check if section contains any mention of pyvista
+    has_plotting = False
+    has_pyvista = any(['pyvista' in line for line in lines])
+    if has_pyvista:
+        for line in lines:
+            if '.plot(' in line or '.show(' in line:
+                has_plotting = True
+    
+    # add directive and indent to entire section
+    if has_plotting:
+        old_lines = lines
+        lines = ['.. pyvista-plot::', '']
+        for line in old_lines:
+            if line:
+                lines.append(f'   {line}')
+            else:
+                lines.append(line)
+
+    if lines:
+        return [header, ''] + lines + ['']
+    else:
+        return [header, '']
+
+# override method
+GoogleDocstring._parse_generic_section = _custom_parse_generic_section
```

Comment:
Playing with private variables is dangerous / fragile. Did you consider doing whatever `matplotlib` does with their directive instead?

https://github.com/matplotlib/matplotlib/blob/c06e8709dde6504d396349c0c80ef019c88c3927/lib/matplotlib/sphinxext/plot_directive.py

Also the current implementation relies on `napoleon`, so will only work for people who use that to document their code (and not, say`numpydoc`).

Reply:
We do exactly what matplotlib did for their directive, it's basically a carbon copy with changes to scrape screenshots rather than matplotlib figures.

This part here is for automating the addition of the rubric when parsing the Examples sections.  Effectively, when it detects that pyvista is used in an examples section, it adds the `pyvista-plot` directive to the section.  Numpydoc does the same thing, which is why you can't use Napoleon and scrape the matplotlib figures; it's literally baked into their source.

As far as I can tell, there's no way of hooking into the section parser.  We could ask for them to add a config value or hook for this, which would make the code much "safer"; I agree that this is a brittle approach.

Reply:
Ahh okay, makes sense. +1 for opening an issue with the napolean folks. If they could add some sort of post-processing hook into the examples section that would be great!

Reply:
Done in https://github.com/sphinx-contrib/napoleon/issues/31.